contributors/devel/strategic-merge-patch.md

```diff
@@ -0,0 +1,205 @@
+Strategic Merge Patch
+=====================
+
+# Background
+
+TODO: @pwittrock complete this section
+
+In the standard JSON merge patch, JSON objects are always merged but lists are
+always replaced. Often that isn't what we want. Let's say we start with the
+following Pod:
+
+```yaml
+spec:
+  containers:
+    - name: nginx
+      image: nginx-1.0
+```
+
+and we POST that to the server (as JSON). Then let's say we want to *add* a
+container to this Pod.
+
+```yaml
+PATCH /api/v1/namespaces/default/pods/pod-name
+spec:
+  containers:
+    - name: log-tailer
+      image: log-tailer-1.0
+```
+
+If we were to use standard Merge Patch, the entire container list would be
+replaced with the single log-tailer container. However, our intent is for the
+container lists to merge together based on the `name` field.
+
+To solve this problem, Strategic Merge Patch uses the go struct tag of the API
+objects to determine what lists should be merged and which ones should not.
+Currently the metadata is available as struct tags on the API objects
+themselves, but will become available to clients as Swagger annotations in the
+future. In the above example, the `patchStrategy` metadata for the `containers`
+field would be `merge` and the `patchMergeKey` would be `name`.
+
+
+# Basic Patch Format
+
+Strategic Merge Patch supports special operations through directives.
+
+There are multiple directives:
+
+- replace
+- merge
+- delete
+- delete from primitive list
+
+`replace`, `merge` and `delete` are mutual exclusive.
+
+## `replace` Directive
+
+### Purpose
+
+`replace` directive indicates that the element that contains it should be replaced instead of being merged.
+
+### Syntax
+
+`replace` directive is used in both patch with directive marker and go struct tags.
+
+Example usage in the patch:
+
+```
+$patch: replace
```

Comment:
Put the example in the context of an actual json struct.  This doesn't show demonstrate how to create a patch with this directive.